Cargo.lock

```diff
@@ -1043,6 +1313,22 @@ version = "0.8.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "5e764a1d40d510daf35e07be9eb06e75770908c27d411ee6c92109c9840eaaf7"
 
+[[package]]
+name = "bitcoin-io"
+version = "0.1.3"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "0b47c4ab7a93edb0c7198c5535ed9b52b63095f4e9b45279c6736cec4b856baf"
+
+[[package]]
+name = "bitcoin_hashes"
+version = "0.14.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "bb18c03d0db0247e147a21a6faafd5a7eb851c743db062de72018b6b7e8e4d16"
+dependencies = [
+ "bitcoin-io",
+ "hex-conservative",
+]
```

Comment:
I'm suss on these bitcoin deps. I don't see why they're needed, as it seems they only get turned on by the `secp256k1` feature of Alloy, which I wouldn't think we'd need?

Reply:
I'll have a crack at it

Reply:
Found the culprit:

https://github.com/alloy-rs/alloy/blob/808b88df4d627e98b3a3ea55ab54d96d264a23bb/crates/consensus/Cargo.toml#L94

Combined with this Cargo bug:

- https://github.com/rust-lang/cargo/issues/10801

So we aren't actually depending on `secp256k1` or these `bitcoin_hashes` crates, but Cargo includes them in the lockfile because of a bug. `cargo tree` _does not_ show these deps (hence the confusing output), but `cargo metadata | jq | less` _does_.

Reply:
Actually I think it's this line, as we're not using the `std` feature of `alloy-consensus`, but we _are_ using the `serde` feature:

https://github.com/alloy-rs/alloy/blob/808b88df4d627e98b3a3ea55ab54d96d264a23bb/crates/consensus/Cargo.toml#L123

Reply:
Yeah I think we just need to live with this.